src/AcceptanceTests/Sending/When_customizing_an_outgoing_native_message.cs

```diff
@@ -33,23 +33,33 @@ public class Context : ScenarioContext
             public bool MessageSessionSentMessageCustomizationReceived { get; set; }
             public bool MessageSessionPublishedMessageCustomizationReceived { get; set; }
             public bool MessageHandlerContextSentMessageCustomizationReceived { get; set; }
+            public bool PhysicalBehaviorMessageSentMessageCustomizationReceived { get; set; }
+            public bool LogicalBehaviorMessageSentMessageCustomizationReceived { get; set; }
             public bool MessageHandlerContextPublishedMessageCustomizationReceived { get; set; }
 
             public bool Completed => MessageSessionSentMessageCustomizationReceived
                                      && MessageSessionPublishedMessageCustomizationReceived
                                      && MessageHandlerContextSentMessageCustomizationReceived
-                                     && MessageHandlerContextPublishedMessageCustomizationReceived;
+                                     && MessageHandlerContextPublishedMessageCustomizationReceived
+                                     && PhysicalBehaviorMessageSentMessageCustomizationReceived
+                                     && LogicalBehaviorMessageSentMessageCustomizationReceived;
         }
 
         public class Endpoint : EndpointConfigurationBuilder
         {
             public Endpoint()
             {
-                EndpointSetup<DefaultServer>();
+                EndpointSetup<DefaultServer>(c=>
+                {
+                    c.Pipeline.Register(typeof(Handler.PhysicalBehavior), "Customizes a native message in a physical behavior");
```

Comment:
This stubbed out but not implemented intentionally?

Reply:
I don't know what you mean with stubbed out

Reply:
I meant that implementation was missing. You [pushed it](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/commit/24c903f4d0f9f6ce4e27ae5b63cfd9a25eb7f1a3#diff-8f76fa64f604b8b31bfc79696b2f01ebc9d7e72883c92a3559ccdcf6226948a1R75) after my comment.

Reply:
ah you mean the pipeline next call?

Reply:
I didn't go to the extend to fully implement it because I wasn't sure after the conversation on the other thread you'd want to take it. Your recent comment made me confident it is a good addition so I finished it.